### PR TITLE
Istioctl: Migrate to galley schemas

### PR DESCRIPTION
**Pull request by nmittler**
This is broken out from #19973

### CODE REVIEW

File: galley/pkg/config/util/pilotadapter/convert.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+// TODO(nmittler): Remove this once pilot migrates to the galley schema model.
+func ConvertConfigToObject(schema collection.Schema, cfg model.Config) (crd.IstioObject, error) {
+	spec, err := gogoprotomarshal.ToJSONMap(cfg.Spec)
+	if err != nil {
+		return nil, err
+	}
+	namespace := cfg.Namespace
+	if namespace == "" {
+		namespace = metav1.NamespaceDefault
+	}
```
Comment (ayj) [inline galley/pkg/config/util/pilotadapter/convert.go]: `if namespace == "" && !schema.ClusterScoped` to support cluster-scoped resources?

File: galley/pkg/config/util/pilotadapter/convert.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		TypeMeta: metav1.TypeMeta{
+			Kind:       schema.Resource().Kind(),
+			APIVersion: schema.Resource().APIVersion(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            cfg.Name,
+			Namespace:       namespace,
+			ResourceVersion: cfg.ResourceVersion,
+			Labels:          cfg.Labels,
+			Annotations:     cfg.Annotations,
```
Comment (ayj) [inline galley/pkg/config/util/pilotadapter/convert.go]: CreationTime and domain also?

File: istioctl/cmd/authz.go [hunk 3]
```diff
@@ -264,6 +267,29 @@ func createAuthorizationPoliciesFromFiles(files []string, rootNamespace string)
 	return authorizationPolicies, nil
 }
 
+func toPilotSchemaSet(s collection.Schemas) schema.Set {
+	all := s.All()
+	out := make(schema.Set, len(all))
```
Comment (ayj) [inline istioctl/cmd/authz.go]:
```suggestion
	out := make(schema.Set, 0, len(all))
```

File: istioctl/cmd/deprecated_cmd.go [hunk 4]
```diff
@@ -44,12 +44,13 @@ import (
 	"istio.io/api/networking/v1alpha3"
 	"istio.io/pkg/log"
 
+	"istio.io/istio/galley/pkg/config/schema/collection"
```
Comment (ayj) [inline istioctl/cmd/deprecated_cmd.go]: We should probably just delete this file / commands. They've been deprecated for more than a year now.

Comment (ayj) [reply istioctl/cmd/deprecated_cmd.go]: @esnible, any objection to removing the deprecated CRUD commands?

File: galley/pkg/config/util/pilotadapter/convert.go (hunk 2 above)
Comment (nmittler) [reply galley/pkg/config/util/pilotadapter/convert.go]: This is a copy of the pilot conversion code: https://github.com/istio/istio/blob/master/pilot/pkg/config/kube/crd/conversion.go#L88.

I've added the creation time ... there is no domain in ObjectMeta, however. Which field were you referring to?

File: galley/pkg/config/util/pilotadapter/convert.go (hunk 1 above)
Comment (nmittler) [reply galley/pkg/config/util/pilotadapter/convert.go]: I think that's right ... I'll make the change. It's not in the original pilot code, however.

File: istioctl/cmd/authz.go (hunk 3 above)
Comment (nmittler) [reply istioctl/cmd/authz.go]: oof ... good catch.

File: galley/pkg/config/util/pilotadapter/convert.go (hunk 2 above)
Comment (ayj) [reply galley/pkg/config/util/pilotadapter/convert.go]: Nevermind about domain.

**Comment by nmittler** [comment]
@ostromart any thoughts on how to debug the validation failure in the linter task? https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/20037/lint_istio/5827/build-log.txt

I tried modifying the [validate unit test](https://github.com/istio/istio/blob/master/istioctl/pkg/validate/validate_test.go#L280) manually, but the failing yaml seems to pass.

**Pull request merged**